lib/full-site-editing/block-templates.php

```diff
@@ -93,18 +98,45 @@ function _gutenberg_get_template_files( $template_type ) {
 				// Subtract ending '.html'.
 				-5
 			);
-			$template_files[] = array(
+			$new_template_item = array(
 				'slug'  => $template_slug,
 				'path'  => $template_file,
 				'theme' => $theme_slug,
 				'type'  => $template_type,
 			);
+
+			if ( 'wp_template_part' === $template_type ) {
+				$template_files[] = _gutenberg_add_template_part_area_info( $new_template_item );
+			} else {
+				$template_files[] = $new_template_item;
+			}
 		}
 	}
 
 	return $template_files;
 }
 
+/**
+ * Attempts to add the template part's area information to the input template.
+ *
+ * @param array $template_info Template to add information to (requires 'type' and 'slug' fields).
+ *
+ * @return array Template.
+ */
+function _gutenberg_add_template_part_area_info( $template_info ) {
+	if ( WP_Theme_JSON_Resolver::theme_has_support() ) {
+		$theme_data = WP_Theme_JSON_Resolver::get_theme_data()->get_template_parts();
+	}
+
+	if ( isset( $theme_data[ $template_info['slug'] ]['area'] ) ) {
+		$template_info['area'] = gutenberg_filter_template_part_area_type( $theme_data[ $template_info['slug'] ]['area'] );
```

Comment:
There's some inconsistency in different parts of the code base for instance in this function name`gutenberg_filter_template_part_area_type` we refer to the "area" as "type" which can be confusing especially since we already have a "type" property on the template and template area objects.

Reply:
Id say here we refer to 'area' as 'area', and 'type' more refers to the different allowed values for that area term.  Perhaps `gutenberg_filter_template_part_area_value` would make more sense?

Reply:
I like value better, but why not just `gutenberg_filter_template_part_area` ?

Reply:
that works too, names are hard 😅

Reply:
https://github.com/WordPress/gutenberg/pull/29679